doc/porting/shields.rst

```diff
@@ -0,0 +1,83 @@
+.. _shields
+
+Shields
+#######
+
+Shields have been introduced on Zephyr as a new element that can be used
+for easier prototyping.
+
+Aim of the shield feature is to provide zephyr formatted shield description
+for easier compatibility with applications.
+
+Shield porting and configuration
+********************************
+
+Shields are available in board directory under :file:`/boards/shields`.
+Following files are provided for shield configuration:
+
+.. code-block:: none
+
+   boards/shields/<shield>
+   ├── Kconfig.shield
+   ├── Kconfig.defconfig
+   ├── <shield>.overlay
+   └── dts.fixup
+
+These files provides shield configuration as follows:
+
+* **Kconfig.shield**: This file defines the shield's config flag
+  (CONFIG_SHIELD_X). This flag will be used in an application to trigger the
```

Comment:
Better to talk in the present rather than the future:

    Applications use this flag to trigger the shield configuration
    on top of the board configuration, enabling each shield
    component or sensor driver.